jetbot/bot.py

```diff
@@ -137,11 +172,23 @@ async def func(ctx: commands.Context):
             if data["issue_tracker"]:
                 message.append("\N{LEFT-POINTING MAGNIFYING GLASS} Issue Tracker: " + self.issue_url(
                     data["issue_tracker"]))
-            channel = await self.product_channel(data["channel_name"], data["category_name"])
-            if channel:
-                message.append("\N{PAGE FACING UP} Discord Channel: " + channel.mention +
-                               ("" if ctx.guild and ctx.guild.id == self.config.guild else " - Join with " +
-                                                                                          self.config.invite))
+            channels = await self.product_channels(data)
+            if channels:
+                if len(channels) == 1:
+                    message.append("\N{PAGE FACING UP} Discord Channel: " + channels[0].mention +
+                                 ("" if ctx.guild and ctx.guild.id == self.config.guild else " - Join with " +
+                                                                                           self.config.invite))
+                else:
+                    message.append("\N{PAGE FACING UP} Discord Channels:")
+                    for channel in channels:
+                        desc = ""
+                        if "channels" in data:
+                            for ch_config in data["channels"]:
+                                if ch_config["name"] == channel.name:
+                                    desc = f" - {ch_config['description']}"
+                                    break
+                        message.append(f"• {channel.mention}{desc}" +
+                                     ("" if ctx.guild and ctx.guild.id == self.config.guild else f" - Join with {self.config.invite}"))
```

Comment:
A similar thought here, anything we can do to have a single branch of logic here rather than two with near-identical logic?